farm-extra-rewards-wrapper/src/wrapper_actions/wrap_farm_token.rs

```diff
@@ -0,0 +1,49 @@
+use crate::single_token_rewards::BaseFarmLogicWrapper;
+
+multiversx_sc::imports!();
+
+#[multiversx_sc::module]
+pub trait WrapFarmTokenModule:
+    auto_farm::whitelists::farms_whitelist::FarmsWhitelistModule
+    + auto_farm::external_storage_read::farm_storage_read::FarmStorageReadModule
+    + rewards::RewardsModule
+    + config::ConfigModule
+    + token_send::TokenSendModule
+    + farm_token::FarmTokenModule
+    + pausable::PausableModule
+    + permissions_module::PermissionsModule
+    + multiversx_sc_modules::default_issue_callbacks::DefaultIssueCallbacksModule
+    + farm_base_impl::base_farm_validation::BaseFarmValidationModule
+    + farm_base_impl::enter_farm::BaseEnterFarmModule
+    + utils::UtilsModule
+    + crate::reward_tokens::RewardTokensModule
+{
+    /// Wraps a single farm token
+    #[payable("*")]
+    #[endpoint(wrapFarmToken)]
+    fn wrap_farm_token_endpoint(&self) -> EsdtTokenPayment {
```

Comment:
why do we need this ?

Reply:
Because users enter with farm tokens, which we "wrap" and return to the user. Then, the user can claim for the base farm, and then claim for this extra rewards farm